### PR TITLE
fix-ts-morph-search-file

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

File: packages/typespec-ts/src/modular/buildSubpathIndex.ts [hunk 1]
```diff
@@ -6,11 +7,14 @@ export function buildSubpathIndexFile(
...
 ) {
-  const apiFiles = project.getSourceFiles(`**/src/${subfolder}/${subpath}/**`);
+  const apiFilePattern = join(srcPath, subfolder, subpath, "**")
+  const apiFiles = project.getSourceFiles(apiFilePattern);
   const indexFile = project.createSourceFile(
     `${srcPath}/${subfolder}/${subpath}/index.ts`
   );
   for (const file of apiFiles) {
+    const filePath = file.getFilePath();
+    console.log(filePath);
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildSubpathIndex.ts:17]:
```suggestion
```

Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildSubpathIndex.ts:17]:
```suggestion
```

File: packages/typespec-test/eng/smoke-test.js [hunk 2]
```diff
@@ -8,6 +8,9 @@ const MAX_BUFFER = 10 * 1024 * 1024;
 function generate(path) {
   let command = `cd ${path} && npx tsp compile ./spec`;
   try {
+    if (existsSync(join(path, "spec", "tspconfig.yaml"))) {
+      command = `cd ${path}/spec && npx tsp compile --config ./tspconfig.yaml .`
```
Comment (MaryGao) [inline packages/typespec-test/eng/smoke-test.js:12]: Then we will lose the entry point selection - the client.tsp or main.tsp

Comment (qiaozha) [reply packages/typespec-test/eng/smoke-test.js:12]: the append client.tsp logic is in the below part.

**Review by MaryGao — APPROVED** [review]
LGTM

**Pull request merged**